### DISCUSSION TITLE
Lesson 6: ganache Local RPC

### DISCUSSION THREAD

**Discussion by oscarmairey** [smartcontractkit/full-blockchain-solidity-course-py#278] [Q&A] [answered]
Hi, 
Once I run the ganache UI on windows and the blockchains creates itself, Brownie does not recognize that it has launched while I type " brownie run .\scripts\deploy.py"  and launches an other ganache while I already have one on my computer instead of using it (for permanant ganache)
"Attached to local RPC client" does not appear

Timestamp: 5:37:00

**Comment by cromewar** [comment, marked as answer]
Hello @Cesarioo  yes this happens because brownie uses the default ganache-cli network if you don't specify another one, so if you want to use your ganache UI network you should add it first to brownie networks.

For example let's suppose your ganache UI looks like this:

![image](https://user-images.githubusercontent.com/20199180/139429643-0843f671-94f0-4ec5-a2a5-fe0f14a2caf5.png)

Yo can add it using:

`brownie networks add Ethereum ganache-local host=http://127.0.0.1:7545 chainid=5777`

Where:

Ethereum: is the netowrk type
ganache-local: is the name, you can name it as you want.
The hos and the chainid comes from ganache as you can see on the image.

So when deploying you must use something like this:

`brownie run scripts/deploy.py --network ganache-local`

**Reply by cromewar** [reply]
Also have in mind you would have closed your terminal and opened again after starting ganachi ui on windows, as it allows the console to recognize the new ganache instance.

**Reply by oscarmairey** [reply]
Hi, thank you for the fast answer
I've done evrything and this pops up
![image](https://user-images.githubusercontent.com/92814105/139459855-2e19d979-339c-44cf-b0ba-27a8a20f21d9.png)

**Reply by cromewar** [reply]
Please copy-paste your deploy script here.

Hint: when you added the network, did you use HTTP for the host or http?

Also please try again without the --network ganache-local but please restart vscode after you have started the ganache ui.

**Reply by Richardwg** [reply]
@Cesarioo 
Hi, Have yo found a solution to this? I'm facing the exact same issue, the deployment activates a local instance of ganache.

**Reply by delphee** [reply]
Same problem here.  Will report back if I find a solution.

**Reply by Richardwg** [reply]
Follow the tutorial for a few more mins and it should work. You need to add a local network as the cromewar suggested.

The instructor actually does this in a few steps and it worked for me. Also don't forget to restart Vs Code after starting ganache.

**Reply by cromewar** [reply]
thanks @Richardwg  I'll mark this question as solved. Best regards

**Reply by sudolink** [reply]
If brownie doesn't auto-find the network you spinned up on the Ganache GUI, you can go to the GUI, find the gear icon(settings), go to server, and set the server's port to 8545. Save & Restart. That way the next time you try to deploy brownie fill find it and attach to it.
But this way, brownie won't remember your deployments, because you didn't explicitly add the network to brownie.

**Reply by Vonnie187** [reply]
does this still import from get_account() ? the local account with the rinkeby wallet address?

**Reply by cromewar** [reply]
Hello @Vonnie187 Could you reformulate for question please? Didn't catch up what you meant.

**Reply by Vonnie187** [reply]
sorry, i'm just having a horrible time getting my  WSL VScode to even recognise ganache Gui , I made another post about it, thank you
